lib/datadog/tracing/contrib/pg/instrumentation.rb

```diff
@@ -125,15 +125,20 @@ def trace(name, sql: nil, statement_name: nil, block: nil)
                 end
 
                 # Read metadata from PG::Result
+                #
+                # It is unclear how result can be `nil` instead of a `PG::Result`
```

Comment:
I don't think we should be commenting code with "we have no idea how this works" types of comments.

When executing a query, the PG gem leverages PQExec [here](https://github.com/ged/ruby-pg/blob/master/ext/pg_connection.c#L1089) (amongst other places). Returning a null pointer is [defined behavior](https://www.postgresql.org/docs/current/libpq-exec.html#LIBPQ-PQEXEC) for PQExec. 

I would suggest updating the comment to mention it is possible to get `nil` as a return, we don't have to explain why in our code.